src/messages.cpp

```diff
@@ -804,16 +807,19 @@ namespace quicr::messages {
     BytesSpan operator>>(BytesSpan buffer, Fetch& msg)
     {
         buffer = buffer >> msg.subscribe_id;
-        buffer = buffer >> msg.track_namespace;
-        buffer = buffer >> msg.track_name;
         msg.priority = buffer.front();
         buffer = buffer.subspan(sizeof(ObjectPriority));
         msg.group_order = static_cast<GroupOrder>(buffer.front());
         buffer = buffer.subspan(sizeof(uint8_t));
+        buffer = buffer >> msg.fetch_type;
+        buffer = buffer >> msg.track_namespace;
+        buffer = buffer >> msg.track_name;
         buffer = buffer >> msg.start_group;
         buffer = buffer >> msg.start_object;
         buffer = buffer >> msg.end_group;
         buffer = buffer >> msg.end_object;
+        buffer = buffer >> msg.joining_subscribe_id;
```

Comment:
Same, only if `kJoniingFetch` will joining_subscribe_id and preceding_group_offset be encoded.